### ISSUE TITLE
Ability to skip tests in GTEST

### ISSUE THREAD

**Issue by ezyang** [pytorch/pytorch#15142]
At the moment, all test skipping in open source in gtest is done as some variation of:

```
  if (skipCondition) return;
```

This is bad, because it means we report skipped tests as "passed". This makes it harder to tell if a test ran or not.

Inside fbcode, we have the convention that:

```
GTEST_FATAL_FAILURE_("Test skipped by client")
```

is not a real fatal failure, but instead means that the test is skipped. But this is just a convention; the TestPilot runner is responsible for reinterpreting this to mean skip, and simply doing this in fbcode would not work.

Newer versions of gtest support skipping but we cannot easily upgrade the version of gtest used in fbcode.

**Comment by ezyang** [comment]
cc @zrphercule

**Comment by zrphercule** [comment]
Yeah newer version of gtest solved this skipping problem perfectly... But this solution also sounds good, thanks!

**Comment by ezyang** [comment]
Based on internal discussion, internal gtest was updated with platform009, so we just have to update the copy in `third_party`
